source/tool/tests/software/mock_helper.py

```diff
@@ -30,6 +32,42 @@ def read_pci_reg(self, bus, device, function, address, size):
         else:
             raise Exception("Unexpected PCI read")
 
+class ACPIHelper(TestHelper):
+    """Generic ACPI emulation
+
+    Emulates an RSDP that points to an RSDT.
+
+    The pointed RSDT does not contain any entry (i.e. does
+    not point to any other table).
+
+    Two regions are defined:
+      * RSDP table [0xE0000, 0xE0020]
+      * RSDT table [0x200, 0x224]
+    """
+    RSDP_DESCRIPTOR = ("RSD PTR " +               # Signature
```

Comment:
Surely you can do a bit better than that. This is not generic enough because you still have to redefine the RSDP in test_acpi_xsdt_list and the RSDT in test_acpi_rsdt_list and test_parse_multi_table. You might want to implement a method to add a table to the RSDT (and adjust its length) without having to redefine it from scratch.

I would also include a default XSDT and make it trivial to switch to which table RSDP is pointing to.